extensions/smallrye-openapi/deployment/src/main/java/io/quarkus/smallrye/openapi/deployment/SmallRyeOpenApiProcessor.java

```diff
@@ -1007,20 +1007,41 @@ private OpenApiDocument storeDocument(OutputTargetBuildItem out,
             OpenAPI staticModel,
             OpenAPI annotationModel,
             List<AddToOpenAPIDefinitionBuildItem> openAPIBuildItems) throws IOException {
+        return storeDocument(out, smallRyeOpenApiConfig, staticModel, annotationModel, openAPIBuildItems, true);
+    }
+
+    private OpenApiDocument storeDocument(OutputTargetBuildItem out,
+            SmallRyeOpenApiConfig smallRyeOpenApiConfig,
+            OpenAPI staticModel,
+            OpenAPI annotationModel,
+            List<AddToOpenAPIDefinitionBuildItem> openAPIBuildItems,
+            boolean includeRuntimeFilters) throws IOException {
 
         Config config = ConfigProvider.getConfig();
         OpenApiConfig openApiConfig = new OpenApiConfigImpl(config);
 
         OpenApiDocument document = prepareOpenApiDocument(staticModel, annotationModel, openAPIBuildItems);
 
-        document.filter(filter(openApiConfig)); // This usually happens at runtime, so when storing we want to filter here too.
+        if (includeRuntimeFilters) {
+            document.filter(filter(openApiConfig)); // This usually happens at runtime, so when storing we want to filter here too.
+        }
+
         // By default, also add the auto generated server
         OASFilter autoServerFilter = getAutoServerFilter(smallRyeOpenApiConfig, true);
         if (autoServerFilter != null) {
             document.filter(autoServerFilter);
         }
-        document.initialize();
 
+        try {
+            document.initialize();
```

Comment:
Just making sure: This doesn't have any side-effects that could be problematic, right?

Reply:
No. Like I mentioned, we actually always had this issue, it's just now being highlighted with the changes to propagate the openapi document in a build item. Before you would have had to have your own filter that used CSI and you would have to configure the document to be stored.

Reply:
Great, thanks for the info!